spec/lib/tasks/remove_students_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rake'
+require 'tempfile'
+
+RSpec.describe 'remove_students', type: :task do
+  describe ':run' do
+    let(:task) { Rake::Task['remove_students:run'] }
+    let(:school) { create(:school) }
+    let(:student_1) { create(:student, school: school) }
+    let(:student_2) { create(:student, school: school) }
+    let(:mock_service) { instance_double(StudentRemovalService) }
+
+    before do
+      # Clear the task to avoid "already invoked" errors
+      task.reenable
+
+      # Mock the confirmation input to avoid interactive prompts
+      allow($stdin).to receive(:gets).and_return("yes\n")
+
+      # Mock StudentRemovalService to avoid actual student removal
+      allow(StudentRemovalService).to receive(:new).and_return(mock_service)
+      allow(mock_service).to receive(:remove_student)
+
+      # Silence console output during tests
+      allow($stdout).to receive(:puts)
+      allow($stdout).to receive(:print)
+      allow(Rails.logger).to receive(:error)
+    end
+
+    describe 'envvar validation' do
+      it 'exits when SCHOOL_ID is missing' do
+        expect { task.invoke }.to raise_error(SystemExit)
+      end
+
+      it 'exits when school is not found' do
+        ENV['SCHOOL_ID'] = 'non-existent-id'
```

Comment:
Use [ClimateControl](https://github.com/thoughtbot/climate_control) to manage `ENV` in testing.